tests/core/utilities/test_encoding.py

```diff
@@ -80,3 +97,122 @@ def test_hex_encode_abi_type(abi_type, value, expected):
 
     actual = hex_encode_abi_type(abi_type, value)
     assert actual == expected
+
+
+@pytest.mark.skipif(sys.version_info.major > 2, reason="these test values only valid for py2")
+@given(
+    st.one_of(st.integers(min_value=0), st.booleans()),
+    st.sampled_from((to_bytes, to_hex, to_decimal)),
+)
+def test_hexstr_if_str_passthrough_py2(val, converter):
+    assert hexstr_if_str(converter, val) == converter(val)
+
+
+@pytest.mark.skipif(sys.version_info.major > 2, reason="these test values only valid for py2")
+@given(
+    st.from_regex(HEX_REGEX),
+    st.sampled_from((to_bytes, to_hex, to_decimal)),
+)
+def test_hexstr_if_str_valid_hex_py2(val, converter):
+    if converter is to_decimal and to_bytes(hexstr=val) == b'':
+        with pytest.raises(ValueError):
+            hexstr_if_str(converter, val)
+    else:
+        assert hexstr_if_str(converter, val) == converter(hexstr=val)
+
+
+@pytest.mark.skipif(sys.version_info.major > 2, reason="these test values only valid for py2")
+@given(
+    st.one_of(st.text(), st.binary()),
+    st.sampled_from((to_bytes, to_hex, to_decimal)),
+)
+def test_hexstr_if_str_invalid_hex_py2(val, converter):
+    try:
+        is_hexstr = (is_hex(val) or val == '')
+    except ValueError:
```

Comment:
*curious*

What's throwing the `ValueError` here?  Something inside `is_hex`?  If so, I think that should be fixed upstream.

Reply:
https://github.com/pipermerriam/ethereum-utils/pull/22